L1Trigger/TrackFindingTracklet/src/PurgeDuplicate.cc

```diff
@@ -681,3 +744,46 @@ std::vector<const Stub*> PurgeDuplicate::getInventedSeedingStub(unsigned int iSe
   }
   return newStubList;
 }
+
+// Tells us the variable bin to which a track would belong
+unsigned int PurgeDuplicate::findVarRInvBin(Tracklet* trk) {
+  std::vector<double> rinvbins = settings_.varrinvbins();
+
+  //Get rinverse of track
+  double rInv = trk->rinv();
+
+  //Check between what 2 values in rinvbins rinv is between
+  auto bins = std::upper_bound(rinvbins.begin(), rinvbins.end(), rInv);
+
+  //return integer for bin index
+  unsigned int rIndx = std::distance(rinvbins.begin(), bins);
+  if (rIndx == std::distance(rinvbins.end(), bins))
+    return rinvbins.size() - 2;
+  else if (bins == rinvbins.begin())
+    return std::distance(rinvbins.begin(), bins);
+  else
+    return rIndx - 1;
+}
+
+// Tells us the overlap bin(s) to which a track belongs
+std::vector<unsigned int> PurgeDuplicate::findOverlapRInvBins(Tracklet* trk) {
+  double rInv = trk->rinv();
+  double overlapsize = settings_.overlapsize();
+  std::vector<double> varrinvbins = settings_.varrinvbins();
+  std::vector<unsigned int> chosenBins;
+  for (long unsigned int i = 0; i < varrinvbins.size() - 1; i++) {
+    if ((rInv < varrinvbins[i + 1] + overlapsize) && (rInv > varrinvbins[i] - overlapsize)) {
+      chosenBins.push_back(i);
+    }
+  }
+  return chosenBins;
+}
+
+// Tells us if a track is in the current bin
+bool PurgeDuplicate::findBin(std::vector<unsigned int> vec, unsigned int num) {
```

Comment:
Both function name and comment are too vague. There's no clear that "bin" is related to invR".

Reply:
Function name has been changed to `isTrackInBin` and comment has been made more explicit.